packages/core/src/snapshot.js

```diff
@@ -323,6 +324,70 @@ function waitForDiscoveryNetworkIdle(page, options) {
 // Used to cache resources across core instances
 const RESOURCE_CACHE_KEY = Symbol('resource-cache');
 
+function* triggerResourceRequests(page, snapshot) {
+  // copy widths to prevent mutation later
+  let widths = snapshot.widths.slice();
+  let initialWidth = widths.shift();
```

Comment:
The mutation this comment is referring to immediately follows (so it's not really later). Also can use rest spread here now which won't mutate the existing array.

```suggestion
  let [initialWidth, ...widths] = snapshot.widths;
```